machinelearning/spark/README.md

```diff
@@ -0,0 +1,112 @@
+# Spark
+
+Apache Spark is a fast, in-memory data processing engine applicable to large-scale data processing, for both batch and streaming machine learning that requires fast access to data sets. Apache Spark’s run-everywhere ethos is now taken a few steps further with the BYOE mentality of Singularity; bring your own environment with mobility of compute.
+
+Spark can be a tricky application to install and configure. But by installing Spark within a Singularity container your installation becomes portable and reproducible, allowing you to spin up instances in the same way on your laptop, in an HPC center or in the cloud. Additionally, since you are using Singularity you can leverage specialized hardware on the underlying hosts like GPUs, or other hardware necessities such as host interconnects, if required by your workload.
+
+We’re going to be making a few assumptions on your setup:
+
+You have a shared home directory across the cluster
+You have passwordless access between nodes on the cluster, even to a single node
+You have a directory in $HOME for Spark related files (i.e. $HOME/spark)
+
+The SIF image can be built like so:
+
+$ cd ~/spark/
+$ sudo singularity build spark.sif spark.def
+
+To finish the setup, it’s necessary to execute a few commands from within the image in our current directory:
+
+$ singularity shell spark.sif
+Singularity spark.sif:~/spark>
+While in the container shell we will run:
+
+> mkdir dropbear
+> cd dropbear/
+> dropbearkey -t rsa -s 4096 -f dropbear_rsa_host_key
+> dropbearkey -t dss -s 1024 -f dropbear_dss_host_key
+> dropbearkey -t ecdsa -s 521 -f dropbear_ecdsa_host_key
+> exit
+This creates dropbear SSH keys, in our current working directory on the host, that will be readable by your user. Without these steps the files in /etc/dropbear cannot be read and the dropbear SSH server will not start.
+
+We will now setup a directory structure to be used by Spark in the containers. Our working directory is $HOME/spark:
```

Comment:
And words like for example `$HOME/spark` (variables, folders, paths, etc.) should be shown between ` quotes of markdown